### ISSUE TITLE
Nested expressions in format call cause it to be an unevaluable expression.

### ISSUE THREAD

**Issue by NPDW** [terraform-linters/tflint#1270]
Hi. I am using the aws-ruleset with a rule of
```hcl
rule "aws_resource_missing_tags" {
  enabled = true

  tags = [
    "Contact",
    "MoreContact",
    "Name",
  ]
```

However, I've noticed that it will not parse this expression.
```hcl
  tags = merge(
    {
      "Name" = format(
        "%s-%s-%s",
        "nat-eip",
        replace(var.vpc_name, "-${var.aws_region}-vpc", ""),
        data.aws_availability_zones.available.names[count.index],
      )
      "Terraform_Module" = format("%s", var.terraform_module)
    },
    var.tags,
  )
```
as invalid. Running TFLint in debug mode I find this error
```
03:04:35 runner_eval.go:73: [INFO] Unevaluable expression found in main.tf:12. TFLint ignores unevaluable expressions.
```

If I then change the above tags map to equal
```hcl
  tags = merge(
    {
      "Name" = format(
        "%s-%s-%s",
        "nat-eip",
       "hi",
        "hello"
      )
      "Terraform_Module" = format("%s", var.terraform_module)
    },
    var.tags,
  )
```
then it will successfully parse and successfully throw an error for not passing the rule.
I've found that any `format` method with methods within it will throw an error of unevaluable.

## Version

```console
$ tflint -v
TFLint version 0.33.1
+ ruleset.aws (0.10.0)

$ terraform -v
Terraform v1.0.10
```

**Comment by bendrucker** [comment]
Data sources are only available at plan time. Any expression that references a data source or resource cannot be evaluated. The expression is the map of tags. Any key or element that is unknown causes the outer expression to be unknown. While in theory it's possible to evaluate some elements that are known, TFLint doesn't do this sort of deep inspection of attributes at the moment.
